### ISSUE TITLE
special character in password failing

### ISSUE THREAD

**Issue by najamsk** [sijms/go-ora#220]
Hello,

we are facing same issue with our passwrod Test#1234 tried using go_ora.BuildUrl with connection string oracle://db_user:Test%[231234@oracle12c.db.com](mailto:231234@oracle12c.db.com):1825/service1 we are able to connect to datbase using sql developer gui tool.

```
urlOptions := map[string]string{
"TRACE FILE": "trace.log",
}
connectionString = go_ora.BuildUrl("oracle12c.db.com", 1825, "service1", "db_user", "Test#1234", urlOptions)
```
When we try running code face following errors, sharing log lines

```
2022/07/07 17:37:40 database from goOra =  oracle://db_user:Test%231234@oracle12c.db.com:1825/service1?TRACE FILE=trace.log
2022/07/07 17:37:40 db.prepare err = no available servers to connect to
```
If i am making a very stupid mistake point it out tried every combination to make this work but still code is in panic.

**Comment by najam-awan-db** [comment]
yes its working correctly instead of service name I was supplying sid. I am moving from mat oci8 driver. oci8 accepts both sid and service name so I was making this mistake. 

choosing this driver was just install driver with go get and no crazy dependencies and complex setups. 

Now i am generate a new binary using your driver and push to all our code environments from staging to production. Facing issue that i reported [#222 ](https://github.com/sijms/go-ora/issues/222)
